src/SFA.DAS.ApprenticeFeedback.Application/Commands/CreateApprenticeFeedback/CreateApprenticeFeedbackHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using MediatR;
+using SFA.DAS.ApprenticeFeedback.Domain.Interfaces;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace SFA.DAS.ApprenticeFeedback.Application.Commands.CreateApprenticeFeedback
+{
+    public class CreateApprenticeFeedbackHandler : IRequestHandler<CreateApprenticeFeedbackCommand, CreateApprenticeFeedbackResponse>
+    {
+        public readonly IApprenticeFeedbackRepository _apprenticeFeedbackRepository;
+
+        public CreateApprenticeFeedbackHandler(IApprenticeFeedbackRepository apprenticeFeedbackRepository)
+        {
+            _apprenticeFeedbackRepository = apprenticeFeedbackRepository;
+        }
+
+
+        public async Task<CreateApprenticeFeedbackResponse> Handle(CreateApprenticeFeedbackCommand request, CancellationToken cancellationToken)
+        {
+
+            // Needs to retrieve apprentice feedback target based on apprentice id which should be in request
+            // May not return a single record. ( currently we assume it returns one )
+            var apprenticeFeedbackTarget = await _apprenticeFeedbackRepository.GetApprenticeFeedbackTargetById(request.ApprenticeId);
+
+            //if i can't find apprentice target record then need to throw an exception as wont be valid feedback
+            if (apprenticeFeedbackTarget == null)
+            {
+                throw new Exception("Apprentice Feedback Target not found.");
+            }
+
+            //validate the attributes supplied -check that the id exists in attribute table if not throw exception
+            var validAttributes = await _apprenticeFeedbackRepository.GetAttributes();
+
+            var allValidAttributes = request.FeedbackAttributes.Select(s => s.Id).All(t => validAttributes.Select(t => t.AttributeId).Contains(t));
+            if(!allValidAttributes)
+            {
+                // throw
+                throw new Exception("Attributes provided are not consistent with those in the database.");
```

Comment:
I'd maybe tweak the message to "Some or all of the attributes supplied to create the feedback record do not exist in the database" 

And while throwing a new Exception is valid, we can be more specific about the type, IE we could use InvalidOperation exception as what is being attempted is incorrect